includes/core/server/request/class-get-charge.php

```diff
@@ -24,7 +24,11 @@ class Get_Charge extends Request {
 	 * @throws Invalid_Request_Parameter_Exception
 	 */
 	protected function set_id( string $id ) {
-		$this->validate_stripe_id( $id, [ 'ch' ] );
+		/**
+		 * `py_XYZ` objects are identical to charges, and sometimes occur
+		 * whenever the payment was made in a non-deposit currency.
+		 */
+		$this->validate_stripe_id( $id, [ 'ch', 'py' ] );
```

Comment:
Are there any other identifiers apart from this?

Reply:
I don't know. Those `py` IDs come from the payment intent object, and if you take a look at [its docs](https://stripe.com/docs/api/payment_intents/object), it only directs to [the charge object](https://stripe.com/docs/api/charges/object), and there is no definition of the prefix there.

This is an additional check on our side, in order to ensure we're not sending invalid objects. Unless there are other reports, I'd assume that those are the only two possible prefixes.